mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py

```diff
@@ -73,28 +83,58 @@ def _state_name(self):
         readonly=True,
         default="NEW"
     )
-    date = fields.Date(
-        'Date',
+
+    state_id = fields.Many2one(
+        'mgmtsystem.nonconformity.state',
+        'State',
+        default=_default_state)
```

Comment:
Why did we replace a Selection list with a Many2one?
For consistency with official modules, you either have a `stage` field with a selection list, on you have Many2one  `stage_id`  (thus using the term Stage instead of State).